src/vcpkg/platform-expression.cpp

```diff
@@ -702,7 +801,7 @@ namespace vcpkg::PlatformExpression
         {
             return false;
         }
-        return Impl{}(lhs.underlying_, rhs.underlying_);
+        return *lhs.underlying_ == *rhs.underlying_;
     }
```

Comment:
This way of comparison makes it easy to introduce bugs because it's easy to forget the dereference and compare pointers.